pyiron/sphinx/base.py

```diff
@@ -1004,117 +1375,71 @@ def compress(self, files_to_compress=None):
     """
 
     def __init__(self):
         self.structure = None
-        self._spin_enabled = False
         self._id_pyi_to_spx = []
         self._id_spx_to_pyi = []
         self.file_dict = {}
 
-    def _odict_to_spx_input(self, element, level=0):
-        """
-        Convert collections.OrderedDict containing SPHInX input
-        hierarchy to string.
-
-        If the item contains:
-            - no value -> considered as flag -> output format: flag;
-            - value is odict
-                -> considered as a group
-                -> output format: group { ...recursive... }
-            - else
-                -> considered as parameter and value
-                -> output format: parameter = value;
-        """
-        line = ""
-        for k, v in element.items():
-            k = k.split("___")[0]
-            if type(v) != list:
-                v = [v]
-            for vv in v:
-                if vv is None:
-                    line += level * "\t" + str(k) + ";\n"
-                elif type(vv) == odict:
-                    if len(vv) == 0:
-                        line += level * "\t" + k + " {}\n"
-                    else:
-                        line += (
-                            level * "\t"
-                            + k
-                            + " {\n"
-                            + self._odict_to_spx_input(vv, level + 1)
-                            + level * "\t"
-                            + "}\n"
-                        )
-                else:
-                    line += level * "\t" + k + " = " + str(vv) + ";\n"
-        return line
-
-    def write_main(
-        self,
-        file_name="input.sx",
-        cwd=None,
-        main_str=None,
-        spin_constraint_enabled=False,
-        enable_kjxc=False,
-        job_name=None,
-    ):
-        """
-        Write the main Sphinx script named input.sx.
+    def copy_potentials(self, potformat="JTH", xc=None, cwd=None):
 
-        Args:
-            file_name (str): name of the file to be written (optional)
-            cwd (str): the current working directory (optinal)
-            main_str (str): the input to write;
-                            if no input is given,
-                            the default input will be written. (optinal)
-        """
-        if main_str is None:
-            self.file_dict['input'] = self.get_main(enable_kjxc=enable_kjxc,
-                                                    spin_constraint_enabled=spin_constraint_enabled,
-                                                    job_name=job_name)
+        if potformat == 'JTH':
+            potentials = SphinxJTHPotentialFile(xc=xc)
+            find_potential_file = find_potential_file_jth
+            pot_path_dict = {"PBE": "jth-gga-pbe"}
+        elif potformat == 'VASP':
+            potentials = VaspPotentialFile(xc=xc)
+            find_potential_file = find_potential_file_vasp
+            pot_path_dict = {"PBE": "paw-gga-pbe", "LDA": "paw-lda"}
```

Comment:
See above: potential format and potential set may not be the same thing.

Reply:
My suggestion is to send `pot_path_dict` up to become an attribute of the InputWriter object itself, so that users can modify it if they want to play with various iterations of e.g. VASP psp's. Its default is an empty dict(), which then gets populated as above.

So to change it,
```python
job = pr.create_job("Sphinx", "example")
job.input_writer.pot_path_dict = {"PBE": "paw-gga-pbe_2018"}
```
Then for that job all PBE calculations should pull potentials from pyiron/resources/vasp/potentials/paw-gga-pbe_2018. I implemented this as of f00eb1ed

Reply:
postponed.